### PR TITLE
TwistSubscriber: add constructor using NodeInterfaces

### PR DESCRIPTION
**Pull request by ottojo**
This adds constructors using NodeInterfaces to TwistSubscriber. This allows use outside of nav2 with non-lifecycle nodes. It would also allow use in places where NodeInterfaces are already used, or places where a node handle is available in any other form than `shared_ptr<LifecycleNode>`.

---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | (none) |
| Primary OS tested on | Ubuntu 24.04 |
| Robotic platform tested on | TBD |
| Does this PR contain AI generated software? | No |
| Was this PR description generated by AI software? | No |

---

## Description of contribution in a few bullet points

* Adds constructors using NodeInterfaces to TwistSubscriber
* Adds NodeInterfaces version of `declare_parameter_if_not_declared` (used in ctor)
* Change all uses to use new constructor, requiring passing the node by reference instead of shared_ptr. I would like to revert that if and when i find a way to make the new and old constructors non-ambiguous. I'm not quite sure why they are, as far as i can tell, only ever one is applicable in any situation.

## Description of documentation updates required from your changes

* [ ] TODO: Check for any doxygen that needs updating

## Description of how this change was tested

* [ ] TODO: Add appropriate unit tests if both constructors are kept.
* [ ] TODO: Add unit test for stamped-only constructor, i think there are none currently.

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in docs.nav2.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists
- [ ] Should this be backported to current distributions? If so, tag with `backport-*`.

### CODE REVIEW

File: nav2_util/src/node_utils.cpp [hunk 1]
```diff
@@ -75,6 +75,18 @@ std::string time_to_string(size_t len)
   return output;
 }
 
+void declare_parameter_if_not_declared(
```
Comment (SteveMacenski) [inline nav2_util/src/node_utils.cpp:78]: Please put this in the new `nav2_ros_common` package where the other parameter utilities are relocated

File: nav2_util/include/nav2_util/twist_subscriber.hpp [hunk 2]
```diff
@@ -79,31 +83,56 @@ class TwistSubscriber
...
+    is_stamped_ = node.get_node_parameters_interface()
+      ->get_parameter("enable_stamped_cmd_vel")
+      .as_bool();
     if (is_stamped_) {
-      twist_stamped_sub_ = node->create_subscription<geometry_msgs::msg::TwistStamped>(
-        topic,
-        qos,
-        std::forward<TwistStampedCallbackT>(TwistStampedCallback));
+      twist_stamped_sub_ =
+        rclcpp::create_subscription<geometry_msgs::msg::TwistStamped>(
```
Comment (SteveMacenski) [inline nav2_util/include/nav2_util/twist_subscriber.hpp:130]: Must use `node->` per https://docs.nav2.org/migration/Kilted.html#new-nav2-ros-common-nav2-lifecycle-node or `nav2_ros_common::interfaces::create_subscription`

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
@ottojo heed the cpplint failing job

**Comment by SteveMacenski** [comment]
@ottojo these comments are more aligned with the `main` version of these rather than Jazzy. What's your intent here for the future? Some of the changes cannot be forward ported past Kilted, so perhaps its worth looking at solutions now while working on this to get these changes in `main` as well so that you can have them in the future as well automatically.

**Comment by mini-1235** [comment]
@ottojo any updates?

**Comment by mini-1235** [comment]
@ottojo I'm going to close this since we are unable to merge it into Jazzy and there hasn't been any recent activity. Please feel free to reopen if you'd like this merged into main.
